### PR TITLE
Improve phpredis support

### PR DESCRIPTION
**Pull request by TheLevti**
The phpredis extension has received many updates since support for it has been added to this library. Some of the features include the ability by the connection to be configured in such a way that all data that is going to redis will be serialized (none, php, igbinary) and/or compressed (none, lzf).

This pull request adds support for compression settings on the phpredis connection.

- Adds support for phpreids compression.
- Adds php7.3 to travis ci
- Updated dev requirements & suggestions.
- Updated suggested min version of the phpredis extension.
- Updated tests to cover serialization and compression options of a phpredis connection.

### CODE REVIEW

**Comment by willemstuursma** [comment]
Thanks @TheLevti. It looks very nice.

I'll review it in a few days. Feel free to ping me here if you think it is taking too long.

**Comment by TheLevti** [comment]
No problem. This is not urgent. Just noticed that issue when I was playing around with phpredis.

File: classes/mutex/PHPRedisMutex.php [hunk 1]
```diff
@@ -55,18 +62,34 @@ protected function add($redisAPI, string $key, string $value, int $expire): bool
...
+    /**
+     * {@inheritDoc}
+     *
+     * @param \Redis|\RedisCluster $redis The Redis or RedisCluster connection.
+     */
     protected function evalScript($redis, string $script, int $numkeys, array $arguments)
     {
-        /** @var Redis $redis */
+        // Determine if we need to compress eval arguments.
+        $lzfCompression = false;
```
Comment (willemstuursma) [inline classes/mutex/PHPRedisMutex.php]: This should be moved into a private method: `hasLzfCompression(Redis $redis): bool`.

Comment (TheLevti) [reply classes/mutex/PHPRedisMutex.php]: Will do that. Will not typehint Redis as we might also get a RedisCluster class. Unfortunately there is no inheritance between those two.

Comment (TheLevti) [reply classes/mutex/PHPRedisMutex.php]: Will typehint it in the docblock.

**Pull request merged**